### PR TITLE
add exist queries support

### PR DESCRIPTION
**Pull request by avarabyeu**

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/database/search/Filter.java [hunk 1]
```diff
@@ -120,4 +120,38 @@ public String toString() {
...
+
+		public FilterBuilder withCondition(FilterCondition condition) {
+			this.conditions.add(condition);
+			return this;
+		}
+
+		public Filter createFilter() {
+			Set<FilterCondition> filterConditions = this.conditions.build();
+			Preconditions.checkArgument(null != target, "Target should not be null");
+			Preconditions.checkArgument(!filterConditions.isEmpty(), "Condition should not be null");
```
Comment (DzmitryKavalets) [inline src/main/java/com/epam/ta/reportportal/database/search/Filter.java]: Incorrect error message

File: src/test/java/com/epam/ta/reportportal/database/search/ConditionBuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.is;
+
+/**
+ * @author Andrei Varabyeu
+ */
+public class ConditionBuilderTest {
+
+	@Test(expected = IllegalArgumentException.class)
```
Comment (DzmitryKavalets) [inline src/test/java/com/epam/ta/reportportal/database/search/ConditionBuilderTest.java]: https://github.com/junit-team/junit4/wiki/exception-testing

File: src/test/java/com/epam/ta/reportportal/database/search/FilterBuilderTest.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+import static org.hamcrest.Matchers.hasSize;
+
+/**
+ * @author Andrei Varabyeu
+ */
+public class FilterBuilderTest {
+
+	@Test(expected = IllegalArgumentException.class)
+	public void checkFilterNoConditions() {
```
Comment (DzmitryKavalets) [inline src/test/java/com/epam/ta/reportportal/database/search/FilterBuilderTest.java]: https://github.com/junit-team/junit4/wiki/exception-testing

File: src/main/java/com/epam/ta/reportportal/database/search/FilterCondition.java [hunk 4]
```diff
@@ -117,4 +121,56 @@ public String toString() {
...
+		}
+
+		public FilterCondition.ConditionBuilder eq(String searchCriteria, String value) {
+			return withCondition(Condition.EQUALS)
+					.withSearchCriteria(searchCriteria)
+					.withValue(value);
+		}
+
+
+		public FilterCondition createCondition() {
```
Comment (DzmitryKavalets) [inline src/main/java/com/epam/ta/reportportal/database/search/FilterCondition.java]: build() ?

File: src/main/java/com/epam/ta/reportportal/database/search/FilterCondition.java [hunk 5]
```diff
@@ -117,4 +121,56 @@ public String toString() {
 				.append(value).append(", searchCriteria = ").append(searchCriteria).append(", negative = ").append(negative).append("}");
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/database/search/FilterCondition.java]: Existing code

File: src/test/java/com/epam/ta/reportportal/database/search/ConditionBuilderTest.java (hunk 2 above)
Comment (avarabyeu) [inline src/test/java/com/epam/ta/reportportal/database/search/ConditionBuilderTest.java]: ??

File: src/main/java/com/epam/ta/reportportal/database/search/Filter.java (hunk 1 above)
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/database/search/Filter.java]: Agreed

File: src/main/java/com/epam/ta/reportportal/database/search/FilterCondition.java (hunk 4 above)
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/database/search/FilterCondition.java]: Aggred

**Pull request merged**